misc/ls-extensions/modules/json-to-record-converter/src/main/java/io/ballerina/converters/JsonToRecordConverter.java

```diff
@@ -191,12 +192,12 @@ public static JsonToRecordResponse convert(String jsonString, String recordName,
                         NodeFactory.createRecordTypeDescriptorNode(recordKeyWord, bodyStartDelimiter,
                                 fieldNodes, null, bodyEndDelimiter);
                 if (isRecordTypeDescriptor) {
-                    typeDefinitionNodeList.add(recordTypeDescriptorNode);
+                    typeDefinitionNodes.put(schema.getKey().trim(), recordTypeDescriptorNode);
```

Comment:
Extract the `schema.getKey().trim()` expression to a separate variable and reuse is

Reply:
Addressed in https://github.com/ballerina-platform/ballerina-lang/pull/34525/commits/9129a28d18a2ad3bae9e19515f274b8ae939c033